test/jdk/jdk/crac/PerfMemoryRestoreTest.java

```diff
@@ -38,52 +41,74 @@
  * @requires os.family == "linux"
  * @library /test/lib
  * @build PerfMemoryRestoreTest
- * @run driver/timeout=30 jdk.test.lib.crac.CracTest
+ * @run driver/timeout=30 jdk.test.lib.crac.CracTest false
+ * @run driver/timeout=30 jdk.test.lib.crac.CracTest true
  */
 
 public class PerfMemoryRestoreTest implements CracTest {
+
+    @CracTestArg(0)
+    boolean perfDisableSharedMem;
+
     @Override
     public void test() throws Exception {
         CracBuilder builder = new CracBuilder();
+        builder.captureOutput(true);
+        if (perfDisableSharedMem) {
+            builder.vmOption("-XX:+PerfDisableSharedMem");
+        } else {
+            builder.vmOption("-XX:-PerfDisableSharedMem");
+        }
         CracProcess checkpoint = builder.startCheckpoint();
         String pid = String.valueOf(checkpoint.pid());
         // This test is run only on Linux where the path is hardcoded
         // in os::get_temp_directory() to /tmp rather than using System.getProperty("java.io.tmpdir")
         Path perfdata = Path.of("/tmp", "hsperfdata_" + System.getProperty("user.name"), pid);
-        long start = System.nanoTime();
-        while (!perfdata.toFile().exists()) {
-            if (System.nanoTime() - start > TimeUnit.SECONDS.toNanos(10)) {
-                throw new IllegalStateException("Perf data file did not appear within time limit in the checkpointed process: " + perfdata);
-            }
-            //noinspection BusyWait
-            Thread.sleep(10);
+        if (perfDisableSharedMem) {
+            Thread.sleep(100);
```

Comment:
I like the refactoring and that we can control this delay independently. 

Nit-picking: should not be this a named constant? And does 100ms has any reason behind, or can it be 1sec (a little bit safer)?

Reply:
Extracted to constant.

My expert opinion says that the almost arbitrary choice of 100 ms is super good enough.